### DISCUSSION TITLE
FAQ - Process, review, and release icons

### DISCUSSION THREAD

**Discussion by LitoMore** [simple-icons/simple-icons#9351] [Help with SVGs]
I will post some FAQs in this discussion.

### How to process a new icon

1. Download the original icon from their homepage
2. Open the icon in Inkscape
3. Change viewbox to 24x24
4. Delete unused shapes
5. Group the remaining shapes
6. Resize this group to 24px with aspect ratio lock
7. Vertical and horizontal center it
8. Save it to a new file and move it to `icons/youricon.svg`
9. Use `npm run add-icon-data` to add metadata to `_data/simple-icons.json`
10. Run `npx svgo icons/youricon.svg`
11. Open `icons/youricon.svg` with a text editor, cleanup unused attributes, and add `<title>youricon</title>`
12. Run `npm run lint` to test it local, it should pass
13. If you encounted a size or center issue while linting, you can retry with 6,7,10,12 steps base on current SVG file
14. Done

#### Here are some useful tips

- [Some workaround for size not exactly in 24px](https://github.com/simple-icons/simple-icons/discussions/9351#discussioncomment-14512041)

### How do we review differences?

1. Download the processed SVG file from the pull request
2. Open the file with Inkscape (or other SVG editor)
3. Drag the original file source icon to Inkscape
4. Resize the original file to 24, and center it, bring it to top
5. Change the fill color of the source one to white or gray
7. Then review the difference
8. Sometimes we also use the [Outline Display Mode](https://www.youtube.com/watch?v=GwcH-LTG1_U) to see the difference

### What is our release schedule?

We usually create a release PR on Sunday, like #12460, and below is the code of our scheduled script:

https://github.com/simple-icons/simple-icons/blob/86ee3f263aa0884c6f3e901172c5efe4e3f32326/.github/workflows/create-release.yml#L6-L8

At that time, we review the PR and merge it. After that our release bot will automatically publish updates to npm and other platforms.

For our icons on [shields.io](https://shields.io), their updates are usually a week behind ours, which means you will see updates on shields.io a week after our release. You can refer to their [how shields.io icon updates scheduled](https://github.com/badges/shields/discussions/5369) for more details.

### Useful tools

- [Simple Icons Companion](https://github.com/LitoMore/simple-icons-companion) - A userscript for GitHub's Simple Icons SVG diff viewer

**Comment by adamrusted** [comment]
```mermaid
flowchart LR
    A(Pick an Issue to work on) --> B{SVG of icon available?}
    B --No--> C[Find highest quality PNG]
    B --Yes--> D{Monochrome SVG present?}
    D --Yes--> E[Follow *How to process a new icon* as posted above]
    D --No--> F[Follow more advanced instructions in the CONTRIBUTING.md file]
    C --> H[Trace the PNG using your preferred vector editing software]
    H --> E
    E --> G
    F --> G
   G[Compare your optimised icon to the source icon] --> I(Submit a PR)
```

**Comment by BPowell76** [comment]
Is there something we are supposed to do about workflows expecting something? Even when I complete all the steps outlined above the pull request shows stuff is pending and needs a maintainer to authorize the workflow.

**Reply by LitoMore** [reply]
It's just because you're a first-time contributor. For those who contribute code for the first time, we need a moderator to start the workflow manually.

**Comment by cscnk52** [comment]
## Some Work Around for Size not exactly in 24px

During image processing, it’s common to encounter SVGs whose size isn’t exactly 24px. This usually happens because the edges are curved, and SVGO optimizes the curvature.

For example, in the Oracle logo, the far-left edge doesn’t have a control point. After optimization with SVGO, the result may no longer be exactly 24px.

As a workaround, I suggest adding a control point at the pole (extreme point), as shown in the video:

![PixPin_2025-09-25_22-19-33](https://github.com/user-attachments/assets/4c461543-2bce-40d3-aa26-540ccc4dfffe)

When adding a control point at extreme point, because this point exist, the size will be exactly in 24px, SVGO will handle correctly. 

see: https://inkscapetutorials.wordpress.com/2014/04/24/inkscape-0-91-feature-insert-node-at-a-curves-extreme-value/

**Reply by LitoMore** [reply]
This is super useful! Thanks for sharing!
